### PR TITLE
Added Cart

### PR DESCRIPTION
**Pull request by HeyNitin**
Added cart components, with the feature to add multiple products and with different quantities.
decrease quantity is currently not working, will have to modify the backend for that.

image

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -3,12 +3,14 @@ import { React } from "react";
 import "./Styles/styles.css";
 import Routes from "./Components/Routes";
 import Navbar from "./Components/Navbar";
+import Footer from "./Components/Footer";
```
Comment (ishitakeshawani) [inline src/App.js:6]: remove unused import

File: src/Components/CartCard.js [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+    const res = await axios.post(
+      "/api/user/cart",
+      { product },
+      {
+        headers: { authorization: token }
+      }
+    );
+    setCart([...res.data.cart]);
+  };
+  // const decreaseQuantity = async () => {
```
Comment (ishitakeshawani) [inline src/Components/CartCard.js:40]: remove commented code from here

File: src/Components/Footer.js [hunk 3]
```diff
@@ -1,5 +1,18 @@
...
+  return (
+    <div className="main-footer">
+      <a
+        href="https://github.com/heynitin/lekhni-ui"
+        target="_blank"
+        rel="noreferrer"
+      >
+        <i class="fa fa-brands fa-github fa-2x"></i>
+      </a>
+      {/* <FontAwesomeIcon icon="fa-brands fa-github" />
```
Comment (ishitakeshawani) [inline src/Components/Footer.js:11]: same here

File: src/Components/Navbar.js [hunk 4]
```diff
@@ -1,9 +1,40 @@
...
+  } = useLength();
+
+  useEffect(() =>
+    (async () => {
+      try {
+        const res = await axios.get("/api/user/wishlist", {
+          headers: { authorization: token }
+        });
+        res && setWishlistLength([...res.data.wishlist].length);
+      } catch (error) {}
```
Comment (ishitakeshawani) [inline src/Components/Navbar.js:24]: console log error  here or  show in on UI

**Review by ishitakeshawani — APPROVED** [review]
Added some suggestions
other things looks good to me

File: src/Components/CartCard.js (hunk 2 above)
Comment (HeyNitin) [reply src/Components/CartCard.js:40]: This code will be there in the next commit and then I'll merge it. just wanted to get it reviewed in the meantime.

File: src/Components/Navbar.js (hunk 4 above)
Comment (HeyNitin) [reply src/Components/Navbar.js:24]: Thanks for pointing it out, I will do it.

File: src/App.js [hunk 5]
```diff
@@ -3,12 +3,14 @@ import { React } from "react";
...
 
 function App() {
   return (
     <div className="App">
       <Navbar />
       <Routes />
+      {/* <Footer /> */}
```
Comment (vaishnavipaliya11) [inline src/App.js:13]: can you remove unwanted code

File: src/backend/controllers/CartController.js [hunk 6]
```diff
@@ -78,12 +89,24 @@ export const removeItemFromCartHandler = function (schema, request) {
...
+    // userCart = userCart.reduce((result, item) => {
+    //   if (item._id === productId && item.qty > 1) {
+    //     return [...result, { ...item, qty: item.qty - 1 }];
+    //   } else if (item._id === productId) {
+    //     return result;
+    //   } else {
+    //     return [...result, item];
+    //   }
+    // }, []);
+
```
Comment (vaishnavipaliya11) [inline src/backend/controllers/CartController.js:109]: can you remove unwanted code

**Review by vaishnavipaliya11 — APPROVED** [review]
Code looks clean.

**Pull request merged**